tiledb/sm/group/group.cc

```diff
@@ -332,7 +354,8 @@ void Group::delete_group(const URI& uri, bool recursive) {
         }
       }
     }
-    storage_manager_->delete_group(uri.c_str());
+    Group::delete_group(
+        uri, storage_manager_->vfs(), storage_manager_->compute_tp());
```

Comment:
We very much don't want to add new references to the `storage_manager_` member variable; the only thing it's necessary for is dispatching to REST. Instead use the `resources_` member variable.

If the old `sm` function is inlined here instead, there's no issue of which arguments to use; there's the `resources_` member variable and the URI function argument.